test/index.test.js

```diff
@@ -1,12 +1,42 @@
 'use strict';
 
 const Hapi = require('hapi');
+const Joi = require('joi');
 const assert = require('chai').assert;
 const jsonapi = require('../');
 const plugin = jsonapi.plugin;
 
 describe('JSON API Plugin', () => {
 
+	describe('Error formatting', () => {
+		it('should format joi errors to more human readable output', async () => {
+			const server = new Hapi.Server({debug: {request: '*'}});
+			await server.register({plugin});
+			server.route({
+				method: 'POST',
+				path: '/test',
+				config: {
+					validate: {
+						failAction: (req, h, err) => err,
+						payload: {
+							fo: Joi.object({
+								bar: Joi.object({
+									banana: Joi.number().required()
+								}).required()
+							}).required()
+						}
+					},
+					handler(req, h) {
+						return h.response().code(204);
+					}
+				}
+			});
+
+			const res = await server.inject({method: 'POST', url: '/test', payload: {fo: {}}});
+			assert.equal(res.result.errors[0].details, 'Validation error: "bar" is required (fo,bar)');
```

Comment:
Shouldn't it be `.` instead of a `,` in `"bar" is required (fo.bar)`? 

At first glance now I read that error message as a missing parameter in a function call or something like that, even if that can't happen during Joi validation.

Reply:
Joi has changed something, dunno know that.. don't want to spend large amounts of time for that either.

Reply:
Fair enough, just thought it was strange to use a `,` there. Hopefully Joi can change it back to something more sensible later on.

Reply:
seems like there was an easy fix for.. new pr coming.